ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.test.tsx

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { ChartsThemeProvider, testChartsTheme } from '@perses-dev/components';
+import { TimeRangeValue } from '@perses-dev/core';
+import { PluginRegistry, useTimeSeriesQueries, TimeRangeContext } from '@perses-dev/plugin-system';
+import { mockPluginRegistryProps, mockTimeSeriesQueryResult } from '../../test';
+import { TimeSeriesChartPanel, TimeSeriesChartProps } from './TimeSeriesChartPanel';
+
+jest.mock('@perses-dev/plugin-system', () => {
+  return {
+    ...jest.requireActual('@perses-dev/plugin-system'),
+    useTimeSeriesQueries: jest.fn(),
+  };
+});
+
+const TEST_TIME_RANGE: TimeRangeValue = { pastDuration: '1h' };
+
+// Test TimeSeriesChart with legend
+const TEST_TIME_SERIES_PANEL: TimeSeriesChartProps = {
+  contentDimensions: {
+    width: 500,
+    height: 500,
+  },
+  spec: {
+    queries: [
+      {
+        kind: 'TimeSeriesQuery',
+        spec: {
+          plugin: {
+            kind: 'PrometheusTimeSeriesQuery',
+            spec: {
+              query: 'rate(caddy_http_response_duration_seconds_sum["5m"])',
+            },
+          },
+        },
+      },
+    ],
+    unit: { kind: 'Decimal', decimal_places: 2 },
+    legend: {
+      position: 'right',
+    },
+  },
+};
+
+describe('TimeSeriesChartPanel', () => {
+  beforeEach(() => {
+    (useTimeSeriesQueries as jest.Mock).mockReturnValue(mockTimeSeriesQueryResult);
```

Comment:
Rather than mocking the hook, why not use the real hook and mock the `PrometheusTimeSeriesQuery` plugin instead? The plugin registry mock utility you added returns a `addMockPlugin` function for just that reason.

Reply:
We discussed this offline, the useTimeSeriesQueries hook has [other runtime dependencies](https://github.com/perses/perses/blob/v0.12.0/ui/plugin-system/src/runtime/time-series-queries.ts#L73-L75) that aren't easy to mock, I'll come back to this later after we work on adding some additional test utils